project-base/storefront/components/Pages/CategoryDetail/CategoryBestsellers/CategoryBestsellers.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button } from 'components/Forms/Button/Button';
+import { ListedProductFragmentApi } from 'graphql/generated';
+import { useGtmSliderProductListViewEvent } from 'gtm/hooks/productList/useGtmSliderProductListViewEvent';
+import { GtmProductListNameType } from 'gtm/types/enums';
+import { useComponentUpdate } from 'hooks/helpers/useComponentUpdate';
+import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { CategoryBestsellersListItem } from './CategoryBestsellersListItem';
+
+const NUMBER_OF_VISIBLE_ITEMS = 3;
+
+type CategoryBestsellersProps = {
+    products: ListedProductFragmentApi[];
+};
+
+export const CategoryBestsellers: FC<CategoryBestsellersProps> = ({ products }) => {
+    const t = useTypedTranslationFunction();
+    const router = useRouter();
+    const [isCollapsed, setIsCollapsed] = useState(false);
+
+    useComponentUpdate(() => setIsCollapsed(false), [router.asPath]);
+    useGtmSliderProductListViewEvent(products, GtmProductListNameType.bestsellers);
+
+    if (products.length === 0) {
+        return null;
+    }
+
+    return (
+        <div className="mb-8">
+            <div className="mb-3 break-words font-bold text-dark lg:text-lg">
+                {t('Do not want to choose? Choose certainty')}
+            </div>
+            <div className="mb-5">
+                {products.map((product, index) => {
+                    return index <= NUMBER_OF_VISIBLE_ITEMS - 1 || isCollapsed ? (
```

Comment:
I am not sure about this but what about prepare/filter data for this loop in advance? I understand that you see here algorithms here and yes it's one small loop more, but to me it feels like correct way since ideally I want to work in the this JSX part with prepared data and don't put logic itself here. Up to you.

Reply:
One more thing which I had on my mind. I thought about separate those filtered items to the outside variable like

```
const shownProducts = products.filter((_, index) => index + 1 <= NUMBER_OF_VISIBLE_ITEMS || !isCollapsed)
```

I'll leave it up to you.